### DISCUSSION TITLE
Tuya Gateway Support for Sub devices

### DISCUSSION THREAD

**Discussion by matheuscanela** [make-all/tuya-local#1960] [Q&A]
Hi Folks,

I am trying to add some Luminance Motion sensor that are paired with my Tuya Gateway. I added all the lights that are wifi based - and worked perfectly, but now I am planning to add the Tuya devices coming from the gateway, is it possible? I can see the sub device id option but it didn't work when I added the gateway as it returns 

> Unable to connect to your device with those details. It could be an intermittent issue, or they may be incorrect.

I tried to add the gateway manually and via the Smart Life cloud-assisted device setup.

See the info below

**Tuya Gateway**
`{
  "result": {
    "active_time": 1714194482,
    "bind_space_id": "2244***",
    "category": "wg2",
    "create_time": 1714194482,
    "custom_name": "External - Tuya Gateway",
    "icon": "smart/icon/ay155***",
    "id": "bf1c3fbafeb8f58***",
    "ip": "125.1***",
    "is_online": true,
    "lat": "***",
    "local_key": "-bkMt.sJ***",
    "lon": "***",
    "model": "DMD2CD",
    "name": "Gateway 2",
    "product_id": "ohwalrf2",
    "product_name": "DMD2CD",
    "sub": false,
    "time_zone": "+10:00",
    "update_time": 1716856910,
    "uuid": "09a8783***"
  },
  "success": true,
  "t": 1716863304204,
  "tid": "f5***6cd41c9911efa664***"
}`

**Luminance Motion sensor connect via gateway**
`{
  "result": {
    "active_time": 1714198435,
    "bind_space_id": "22***",
    "category": "pir",
    "create_time": 1714198435,
    "custom_name": "External - Front Near Driveway",
    "icon": "smart/icon/ay153198553700***",
    "id": "bff297e78c***",
    "ip": "",
    "is_online": true,
    "lat": "***",
    "local_key": "",
    "lon": "***",
    "model": "ZG-204L",
    "name": "Luminance Motion sensor  4",
    "product_id": "3towulqd",
    "product_name": "Luminance Motion sensor",
    "sub": true,
    "time_zone": "+10:00",
    "update_time": 1716856348,
    "uuid": "bff297e78c0a***"
  },
  "success": true,
  "t": 1716864570863,
  "tid": "e***41e1f1c9c11ef96b8***"
}`

I also enabled the debug - see the log attached
[home-assistant_tuya_local_2024-05-28T02-55-09.914Z.zip](https://github.com/make-all/tuya-local/files/15463383/home-assistant_tuya_local_2024-05-28T02-55-09.914Z.zip)

And I cannot see a gateway under Choose gateway if I use the wizard
![Screenshot 2024-05-28 125501](https://github.com/make-all/tuya-local/assets/6635137/d3390065-59fc-4cd9-be11-acc41be4a4e2)

**Comment by make-all** [comment]
It seems the hubs are recognised by the fact that they do not have local keys.  But for some hubs, there are local functions as well, so they will have local keys.  I have added some additional checks based on device category, to pull out known hub categories regardless of whether they have a local key or not.
